pkg/registry/core/pod/storage/eviction.go

```diff
@@ -220,7 +221,7 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		pdbName = pdb.Name
 
 		// If the pod is not ready, it doesn't count towards healthy and we should not decrement
-		if !podutil.IsPodReady(pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
+		if !podutil.IsPodReady(pod) && enoughHealthyReplicas(pdb) {
 			updateDeletionOptions = true
```

Comment:
this is `requireCurrentPodResourceVersionForDeletion` and it is set in cases when the current pod state is being used to justify allowing the pod to be evicted instead of the more usual case where the pod state was not used to justify deletion?

Reply:
Correct.  We want to ensure this pod doesn't become ready and another actor makes a different decision at the same (very close) time we are deleting it.  `requireCurrentPodResourceVersionForDeletion` is a good description of what we're doing.

Reply:
let's update the name to `requireCurrentPodResourceVersionForDeletion` I think